src/xpath.c

```diff
@@ -5640,7 +5650,8 @@ moveto_node(struct lyxp_set *set, struct lyd_node *cur_node, struct lys_module *
                     }
                     ++i;
                 } else if (ret == EXIT_FAILURE) {
-                    lydict_remove(ctx, name_dict);
+                    if (name_dict)
```

Comment:
Again missing `{}` but you do not even need to test for `NULL` (same problem 3x in the function).